py-polars/tests/unit/operations/test_explode.py

```diff
@@ -405,14 +405,14 @@ def test_fast_explode_merge_left_16923() -> None:
 @pytest.mark.parametrize(
     ("values", "exploded"),
     [
-        (["foobar", None], ["f", "o", "o", "b", "a", "r", ""]),
-        ([None, "foo", "bar"], ["", "f", "o", "o", "b", "a", "r"]),
+        (["foobar", None], ["f", "o", "o", "b", "a", "r", None]),
```

Comment:
Revert test case changed by https://github.com/pola-rs/polars/pull/18984/files#diff-2416ad1b63621571475db2cc59594c85383cf09e30739703a5217b9cff496778R408-R415

We have a unintended behavior change in `str.explode` on the current release, but it's not too critical given the functionality is already deprecated.

Interesting note - the reason why `eq_missing` broke this is because `str.explode` is defined as `F.when(split.ne_missing([])).then(split).otherwise([""]).explode()` (https://github.com/pola-rs/polars/blob/main/py-polars/polars/expr/string.py#L2345)

Reply:
Can we get the same result with a different query? We're free to change this - but we should guarantee the output remains the same.

Reply:
This issue changed `str.explode` to coalesce NULLs to empty string -

```python
s = pl.Series(["ab", None, "cd"])
df = s.to_frame()

split = pl.first().str.split("")
print(s)
print(df.select(pl.when(split.ne_missing([])).then(split).otherwise([""]).explode()))
```

Release 1.9.0
```python
shape: (3,)
Series: '' [str]
[
	"ab"
	null
	"cd"
]
shape: (5, 1)
┌─────┐
│     │
│ --- │
│ str │
╞═════╡
│ a   │
│ b   │
│     │
│ c   │
│ d   │
└─────┘
```

Release 1.8.0
```python
shape: (3,)
Series: '' [str]
[
        "ab"
        null
        "cd"
]
shape: (5, 1)
┌──────┐
│      │
│ ---  │
│ str  │
╞══════╡
│ a    │
│ b    │
│ null │
│ c    │
│ d    │
└──────┘
```

This PR restores the behavior to 1.8.0

Reply:
> Can we get the same result with a different query? We're free to change this - but we should guarantee the output remains the same. I wrote this to mimic the now-removed string explode behavior.

I can maybe do `when((split.list.len() > 0) | split.is_null())`?

Reply:
Ah, I understand :+1: carry on then! :smile: